python/lsst/pipe/base/pipeline_graph/visualization/_options.py

```diff
@@ -0,0 +1,103 @@
+# This file is part of pipe_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ("NodeAttributeOptions",)
+
+import dataclasses
+from typing import Literal
+
+
+@dataclasses.dataclass
+class NodeAttributeOptions:
+    """Struct holding options for how to display and possibly merge nodes."""
+
+    dimensions: Literal["full"] | Literal["concise"] | Literal[False] | None
+    """Options for displaying dimensions.
+
+    Possible values include:
+
+    - ``"full"``: report fully-expanded dimensions.
+    - ``"concise"``: report only dimensions that are not required or implied
+      dependencies of any reported dimension.
+    - `False`: do not report dimensions at all.
+    - `None`: context-dependent default behavior.
+    """
```

Comment:
@leeskelvin, here's part of the answer to your question about which dimensions are shown: the default is "concise".